src/ng/http.js

```diff
@@ -1135,29 +1135,86 @@ function $HttpProvider() {
     }
 
 
-    function buildUrl(url, params) {
-      if (!params) return url;
-      var parts = [];
-      forEachSorted(params, function(value, key) {
-        if (value === null || isUndefined(value)) return;
-        if (!isArray(value)) value = [value];
-
-        forEach(value, function(v) {
-          if (isObject(v)) {
-            if (isDate(v)) {
-              v = v.toISOString();
-            } else {
-              v = toJson(v);
-            }
-          }
-          parts.push(encodeUriQuery(key) + '=' +
-                     encodeUriQuery(v));
-        });
-      });
-      if (parts.length > 0) {
-        url += ((url.indexOf('?') == -1) ? '?' : '&') + parts.join('&');
+    function buildUrl(url, params, mode) {
+      params = $httpUrlParams.serialize(params, mode);
+      if (params) {
+        url += ((url.indexOf('?') == -1) ? '?' : '&') + params;
       }
       return url;
     }
   }];
 }
+
+/**
+ * @ngdoc provider
+ * @name $httpUrlParamsProvider
+ * @description
+ * Use `$httpUrlParamsProvider` to change the default behavior of the {@link ng.$httpUrlParams $httpUrlParams} service.
+ * */
+function $HttpUrlParamsProvider() {
+
+  var paramSerializers = createMap();
+
+  function serializeParams(params, addArrayMarker) {
+    var parts = [];
+
+    if (!params) return '';
```

Comment:
Pretty minot, but why not move this to the top (to avoid unused array creation) ?